libs/core/algorithms/include/hpx/parallel/algorithms/contains.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/execution/algorithms/detail/predicates.hpp>
+#include <hpx/executors/execution_policy.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/parallel/algorithms/detail/dispatch.hpp>
+#include <hpx/parallel/algorithms/detail/distance.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+#include <hpx/parallel/util/loop.hpp>
+#include <hpx/type_support/identity.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+
+namespace hpx::parallel {
+    namespace detail {
+        struct contains : public algorithm<contains, bool>
+        {
+            constexpr contains() noexcept
+              : algorithm("contains")
+            {
+
+            }
+
+            template <typename ExPolicy, typename Iterator, typename Sentinel,
+                typename T, typename Proj>
+            static constexpr bool sequential(ExPolicy, Iterator first,
+                Sentinel last, const T& val, Proj&& proj)
+            {
+                auto const distance = detail::distance(first, last);
+                if (distance <= 0)
+                    return false;
+
+                const auto itr = util::loop_pred<
+                    std::decay_t<hpx::execution::sequenced_policy>>(first, last,
+                    [&val, &proj] (const auto& cur)
+                    { return HPX_INVOKE(proj, *cur) == val; });
+
+                return itr != last;
+            }
+
+            template <typename ExPolicy, typename Iterator, typename Sentinel,
+                typename T, typename Proj>
+            static util::detail::algorithm_result_t<ExPolicy, bool> parallel(
+                ExPolicy&& policy, Iterator first, Sentinel last, const T& val,
+                Proj&& proj)
+            {
+                const auto distance = detail::distance(first, last);
+                if (distance <= 0)
+                {
+                    return util::detail::algorithm_result<ExPolicy, bool>::get(
+                        false);
+                }
+
+                const auto itr = util::loop_pred<
+                    std::decay_t<ExPolicy>>(first, last, [&val, &proj]
+                    (const auto& cur) { return HPX_INVOKE(proj, *cur) == val; });
+
+                if (itr == last)
+                {
+                    return util::detail::algorithm_result<ExPolicy, bool>::get(
+                        false);
+                }
+
+                return util::detail::algorithm_result<ExPolicy, bool>::get(true);
+            }
```

Comment:
This is probably work-in-progress, but I just want to mention that util::loop_pred doesn't actually use the execution policy to run in parallel, so this will not run in multiple threads. You'll have to use a partitioner, which splits the work and launches it on multiple (the [`equal` algorithm](https://github.com/STEllAR-GROUP/hpx/blob/3521f71bd2ccd04e370ac2181bb7b6987192704c/libs/core/algorithms/include/hpx/parallel/algorithms/equal.hpp#L593-L633) is a a nice example).

Reply:
Ah I was unaware at the time I wrote that, I should have been more mindful of what I was using. I checked out equal algorithm and I made some changes that now use the partitioner in the parallel. Also, I was hesitant to do this but I created a file in the parallel/algorithms/detail called contains.hpp, holding the implementation details for the algorithm.